[PresleyChew] iP

PresleyChew opened the pull request:
# Gutti

> The smallest feline is a masterpiece - Leonardo da Vinci [Source](https://www.laphamsquarterly.org/roundtable/old-master-cats#:~:text=One%20great%20artist%20who%20did,a%20feature%20of%20a%20painting.)

Gutti is the all you need for task managing! It's,
- text-based
- easy to use
- ~~AWESOME~~  *great* to use

All you need to do is,
1. Download
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders

Here is the `Main` method:

```ruby
    public static void main(String[] args) {
        new Gutti().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

PresleyChew committed:
Level 0. Rename, Greet, Exit

PresleyChew committed:
Level 1. Echo

PresleyChew committed:
Level 2. Add, List

PresleyChew committed:
Added Javadoc comments.

PresleyChew committed:
Level 3. Mark as Done
Added new class Task
Added try catch block for edge cases
Will add Javadocs comments in next commit.

PresleyChew committed:
Added Javadocs comments for remaining methods, updated javadocs for echo method in Gutti.java

PresleyChew committed:
Level 4. ToDos, Events, Deadlines
Created 3 new class that inherits from Task created in Level 3. Todo,Event,Deadline.
Added Javadocs.
Made code more readable and compact in Echo method under class Gutti via abstraction.

PresleyChew committed:
Test using the I/O redirection technique

PresleyChew committed:
Level 5. Handle Errors
Created GuttiException Class
Will add javadocs comments in next commit
Handled errors and provided user with instruction on how to correct it.

PresleyChew committed:
Added Javadocs comments

PresleyChew committed:
Level 6. Delete
added more exceptions to handle potential edge cases
reimplemented list using arraylist for dynamic adding and removing
added necessary javadoc comments.

PresleyChew committed:
Add Save Feature

PresleyChew committed:
Merge branch 'branch-Level-7'

PresleyChew committed:
Add support for Dates and Times

PresleyChew committed:
Merge branch 'branch-Level-8'

PresleyChew committed:
Add more OOP

PresleyChew committed:
Packed into Package

PresleyChew committed:
Merge branch 'add-gradle-support' of https://github.com/PresleyChew/ip

PresleyChew committed:
Add JUnit Tests

PresleyChew committed:
Create JAR

PresleyChew committed:
Add manifest for Jar File

PresleyChew committed:
bug fix: parsed todo

PresleyChew committed:
Add Javadoc

PresleyChew committed:
Merge branch 'branch-A-JavaDoc'

PresleyChew committed:
Adjust code format

PresleyChew committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Gutti/Deadline.java

Resolution: branch-A-JavaDoc's JavaDoc to be included

PresleyChew committed:
Add Find Feature

PresleyChew committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/gutti.txt

Resolution : Deleted data in gutti.txt due to mistake in commiting data file.

seandias commented on src/main/java/Gutti/FindCommand.java:
```diff
@@ -0,0 +1,74 @@
+ * and output tasks that match the search criteria.
+ * </p>
+ */
+public class FindCommand extends Command{
```
Good naming conventions used  and JavaDoc used where appropriate. Code is concise and to the point.

seandias commented on src/main/java/Gutti/FindCommand.java:
```diff
@@ -0,0 +1,74 @@
+            }
+        }
+        if (!integerList.isEmpty()) {
+            System.out.println("____________________________________________________________");
```
Could put the line into a variable  instead of always having to print the line as the line is always the same.

seandias reviewed (commented):
Just some minor changes such as line can be put in a variable to make it more reader friendly. Overall following coding styles and JavaDocs used well. Nice one!

PresleyChew committed:
Conduct CheckStyle

PresleyChew committed:
Merge branch 'branch-CheckStyle'

PresleyChew committed:
Add support for GUI

PresleyChew committed:
Merge branch 'branch-Level-10'

reginateh commented on src/main/java/gutti/Command.java:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Represents an abstract command that can be executed.
+ */
+public abstract class Command {
```
I like that you created an abstract class for commands.

reginateh commented on src/main/java/gutti/DateTimeUtil.java:
```diff
@@ -0,0 +1,54 @@
+ * using various predefined formats. This class provides methods to attempt parsing
+ * with multiple date formats and returns a {@link LocalDateTime} if successful.
+ */
+public class DateTimeUtil {
```
I think this util file makes your methods very organized.

reginateh commented on src/main/java/gutti/DateTimeUtil.java:
```diff
@@ -0,0 +1,54 @@
+        try {
+            return LocalDateTime.parse(dateTimeString, FORMAT1);
+        } catch (DateTimeParseException e) {
+            // Continue to next formatter
```
I like how you used comments to explain empty blocks.

reginateh commented on src/main/java/gutti/DeadlineCommand.java:
```diff
@@ -0,0 +1,59 @@
+/**
+ * Represents a command to add a Deadline task.
+ */
+public class DeadlineCommand extends Command {
```
I like that you created a class for each type of command.

reginateh reviewed (commented):
Overall very clean code!

PresleyChew committed:
The current code does not verify assumption at runtime.

Let's add assertions at the relevant checkpoints to ensure
verification at checkpoint.

By using assertions we will be able to ensures verification of
assumptions during runtime.

PresleyChew committed:
Currently loadTasksFromFile under Storage is too lengthy.

To allow us to find potential bugs easier it is important for us to
shorten the code.

Let's use the Extract Function to shorten the code.

PresleyChew committed:
Merge pull request #1 from PresleyChew/branch-A-Assertions

The current code does not verify assumption at runtime.

PresleyChew committed:
Merge branch 'master' of https://github.com/PresleyChew/ip

PresleyChew committed:
Merge branch 'branch-A-CodeQuality'

PresleyChew committed:
Add Stats command. Allowing users to type "stats" to see the number
of tasks completed in the past week.

PresleyChew committed:
Update README.md

PresleyChew committed:
Amended typo

PresleyChew committed:
Merge branch 'master' of https://github.com/PresleyChew/ip

PresleyChew committed:
Add Javadocs for StatsCommand

PresleyChew committed:
Add JavaDocs for MainWindow

PresleyChew committed:
Add JavaDocs for DialogBox

PresleyChew committed:
Amend small portion of code to improve code quality

PresleyChew committed:
Debug for problems associated with stat feature

PresleyChew committed:
Add new method that aids in creation of gutti.txt file if does not
exist

PresleyChew committed:
Add readME and fix exit command

PresleyChew committed:
Update README.md
